code/game/machinery/computer/teleporter.dm

```diff
@@ -93,6 +93,11 @@
 			addtimer(CALLBACK(src, .proc/finish_calibration), 50 * (3 - power_station.teleporter_hub.accuracy)) //Better parts mean faster calibration
 			. = TRUE
 
+/obj/machinery/computer/teleporter/proc/set_teleport_target(new_target)
+	var/old_target = target
+	target = new_target
+	SEND_SIGNAL(src, COMSIG_TELEPORTER_NEW_TARGET, target, old_target)
```

Comment:
```suggestion
/obj/machinery/computer/teleporter/proc/set_teleport_target(new_target)
	. = target
	SEND_SIGNAL(src, COMSIG_TELEPORTER_NEW_TARGET, new_target)
	target = new_target
```
Try to send the signal as soon as you can, so in the future if something wants to return something that cancels it that can be done with less effort.
The old target value can be deduced since the source is always passed, and redefining procs that call parent before will get the old value through the return, like in other similar setter procs.